File: flutter/test/features/pets/preconsult_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:petsfollow_mobile/core/models/preconsult.dart';
import 'package:petsfollow_mobile/core/models/visit.dart';

void main() {
  test('PreconsultAnswers.toJson omits empty comment', () {
    const a = PreconsultAnswers(
      chiefComplaint: ' Vomiting ',
      duration: 'few_days',
      behavior: 'lethargic',
      appetite: 'decreased',
      thirst: 'normal',
      elimination: 'normal',
      urgency: 'high',
    );
    final json = a.toJson();
    expect(json['chiefComplaint'], 'Vomiting');
    expect(json.containsKey('comment'), isFalse);
    expect(json['urgency'], 'high');
  });

  test('Visit.preconsultPending requires confirmed + pending', () {
    final pending = Visit.fromJson({
      'id': 'v1',
      'petId': 'p1',
      'status': 'confirmed',
      'preconsultStatus': 'pending',
    });
    expect(pending.preconsultPending, isTrue);
    final submitted = Visit.fromJson({
      'id': 'v1',
      'petId': 'p1',
      'status': 'confirmed',
      'preconsultStatus': 'submitted',
    });
    expect(submitted.preconsultPending, isFalse);
  });

  test('PreconsultIntake.fromJson parses answers', () {
    final in_ = PreconsultIntake.fromJson({
      'id': 'i1',
      'visitId': 'v1',
      'status': 'submitted',
      'petName': 'Bella',
      'answers': {
        'chiefComplaint': 'Cough',
        'duration': 'week',
        'behavior': 'normal',
        'appetite': 'normal',
        'thirst': 'increased',
        'elimination': 'unknown',
        'urgency': 'low',
        'comment': 'night only',
      },
    });
    expect(in_.isSubmitted, isTrue);
    expect(in_.petName, 'Bella');
    expect(in_.answers.thirst, 'increased');
    expect(in_.answers.comment, 'night only');
  });
}

File: go/internal/store/preconsult_test.go
package store

import "testing"

func TestValidatePreconsultAnswers(t *testing.T) {
	ok := PreconsultAnswers{
		ChiefComplaint: "Vomiting",
		Duration:       "few_days",
		Behavior:       "lethargic",
		Appetite:       "decreased",
		Thirst:         "normal",
		Elimination:    "normal",
		Urgency:        "medium",
		Comment:        "since Tuesday",
	}
	if err := ValidatePreconsultAnswers(ok); err != nil {
		t.Fatalf("expected ok, got %v", err)
	}
	bad := ok
	bad.ChiefComplaint = ""
	if err := ValidatePreconsultAnswers(bad); err == nil {
		t.Fatal("expected chief_complaint_required")
	}
	bad = ok
	bad.Duration = "forever"
	if err := ValidatePreconsultAnswers(bad); err == nil {
		t.Fatal("expected invalid_duration")
	}
	xss := ok
	xss.ChiefComplaint = "<script>alert(1)</script>"
	if err := ValidatePreconsultAnswers(xss); err == nil || err.Error() != "html_not_allowed" {
		t.Fatalf("expected html_not_allowed, got %v", err)
	}
	xss = ok
	xss.Comment = "ok <b>no</b>"
	if err := ValidatePreconsultAnswers(xss); err == nil || err.Error() != "html_not_allowed" {
		t.Fatalf("expected html_not_allowed on comment, got %v", err)
	}
}
